### ISSUE TITLE
camelcase check for properties seems broken

### ISSUE THREAD

**Issue by Means88** [eslint/eslint#9715]
**Tell us about your environment**

* **ESLint Version: 4.13.1**
* **Node Version: 6.9.1**
* **npm Version: 3.10.8**

**What parser (default, Babel-ESLint, etc.) are you using?**
default

<details>
<summary>Configuration</summary>
Written in command line directly as below.
</details>

```js
const foo = { a_b: '' };
const { a_b } = foo;
```

```bash
node_modules/.bin/eslint --rule 'camelcase: [2, { "properties": "never" }]' --parser-options=ecmaVersion:6 1.js
```

**What did you expect to happen?**
exit with 0

**What actually happened? Please include the actual, raw output from ESLint.**
```
2:9  error  Identifier 'a_b' is not in camel case  camelcase
```

**screenshots**

![image](https://user-images.githubusercontent.com/5810331/33878686-1e7c7560-df67-11e7-83c0-6bf9c43fc6f3.png)

It will not throw an error before `@4.13`

![image](https://user-images.githubusercontent.com/5810331/33878793-67c209ba-df67-11e7-92d5-d009e79af8ea.png)

**Comment by platinumazure** [comment]
This is working as intended (as of the bugfix for issue #9468).

This line:

```js
const { a_b } = foo;
```

Is equivalent to this line:

```js
const a_b = foo.a_b;
```

The right-hand side should not be reported with `properties: never`, but the left-hand side should be (because you're declaring a new variable, not an object property).

You can still get at properties in an object through destructuring-- they just need to be reassigned to a camelcase variable:

```js
const { a_b: aB } = foo;
```

**Comment by Means88** [comment]
@platinumazure 👍 Yes, I think you are right.
